blocks/copilot/copilot.js

```diff
@@ -105,76 +97,81 @@ export default function decorate(block) {
-    event.preventDefault(); // Prevent form submission
-	
-	document.getElementById("container").style.display = "none"; 
-	document.getElementById("result").style.display = "block";
-	document.getElementById("buttons").style.display = "none"; 
-	
-	const myInterval = setInterval(function() {
-		document.getElementById("loading").innerHTML = document.getElementById("loading").innerHTML + "..."; 
-	}, 200);
-	
-	const myInterval2 = setInterval(function() {
-		document.getElementById("loading").innerHTML = "AI at work"; 
-	}, 2000);
-	
-	document.getElementById('regenerate').onclick = function() {
+
+	document.getElementById("copy").addEventListener("click", async function (event) {
+		const data = [new ClipboardItem({ [clipboardData.type]: clipboardData })];
+		navigator.clipboard.write(data);
+	});
+
+	document.getElementById('regenerate').onclick = function () {
 		document.getElementById('myForm').submit();
 	};
-	
-	
-    // Get the input value
-	const promptStr = document.getElementById('prompt').value;
-    //Prepare chat api
-	await initAzureAPI(cfg);
-	//Call chat api
-	const jsonRes = await callAzureAPI(/*"Design a brochure on hiking"*/promptStr, cfg, "user");
-	console.log(jsonRes);
-	
-	var design = templates[jsonRes.template];
-	var preview_html = "";
-		
-	for(var i = 0; i < design.length; i++) {
-		var block_name = Object.keys(design[i])[0];
-		
-		console.log("Generating " + jsonRes.template);
-		console.log("Topic " + jsonRes.topic);
-		console.log("Block " + design[i][block_name]);
-		
-		const content = await generate(jsonRes.topic, design[i][block_name], cfg);
+
+	document.getElementById("container").style.display = "flex";
+	document.getElementById("container").style["flexDirection"] = "column";
+	document.getElementById("container").style["textAlign"] = "center";
+	document.getElementById("result").style.display = "none";
+
+	document.getElementById('myForm').addEventListener('submit', async function (event) {
+		event.preventDefault(); // Prevent form submission
+		document.getElementById("container").style.display = "none";
+		document.getElementById("result").style.display = "block";
+		document.getElementById("buttons").style.display = "none";
+
+		const myInterval = setInterval(function () {
+			document.getElementById("loading").innerHTML = document.getElementById("loading").innerHTML + "...";
+		}, 200);
+
+		const myInterval2 = setInterval(function () {
+			document.getElementById("loading").innerHTML = "AI at work";
+		}, 2000);
+
+
+		// Get the input value
+		const promptStr = document.getElementById('prompt').value;
+		//Prepare chat api
+		console.log("[copilot] Initiate chat with Azure to train it to parse prompt to JSON");
+		const conversation = await initiateChat();
+		console.log("[copilot] Chat initiated");
+		//Call chat api
+		conversation.push({
+			role: 'user',
+			content: `Parse and return only the JSON representation for "Create ${promptStr}" without anything else`,
```

Comment:
@sdmcraft Does prefixing Create before prompt work well, if prompt is "Build a website...."?

Reply:
yes probably "Create..." is not needed as we don't use the `action` for now anyways. I'll remove it.